torch/csrc/jit/graph_executor.cpp

```diff
@@ -409,70 +412,117 @@ struct GraphExecutorImpl {
     return false;
   }
 
-
-  // remove ReplaceIfUndef(v, replacement) nodes that consume inputs with 'v' if
-  // the input is defined, and 'replacement' if it is not.
-  // Note: this is a very limited pass. It looks at undefined inputs,
-  // and cleans up ReplaceIfUndef nodes inserted by autodiff.
+  // propagate undefined information through a gradient graph
+  // Note: this is a very limited pass. It only propagates undefines for
+  // operations generated by the symbolic autodiff code and cleans up
+  // AutogradAdds when possible. Outputs of other nodes are conservatively
+  // marked MaybeDefined and not optimized.
   void specializeUndef(Graph & g, const ArgumentSpec & spec) {
+    enum class State { Defined, Undefined, MaybeDefined };
```

Comment:
nit: Wouldn't it be clearer to say `Unknown` instead of `MaybeDefined` (which seems to lean towards the `Defined` option)?